gradle/lib/dependabot/gradle/file_fetcher.rb

```diff
@@ -41,7 +41,13 @@ def all_buildfiles_in_build(root_dir)
         files += subproject_buildfiles(root_dir)
         files += dependency_script_plugins(root_dir)
         files + included_builds(root_dir).
-                flat_map { |dir| all_buildfiles_in_build(dir) }
+                flat_map do |dir|
+                  all_buildfiles_in_build(dir)
+                rescue Dependabot::DependencyFileNotFound
+                  # included build can be in a submodule
+                  nil
```

Comment:
Just realized this is going to catch _all_ failures to find a file, which seems overly broad.

Is there a way to definitively identify that it's in a git submodule? (since it sounds like Gradle Submodules refer to files in git submodules?)

You pointed out the `file_exists_in_submodule?` method, could we just use that here too?
https://github.com/dependabot/dependabot-core/blob/4cab0ee282f9d774e65bad0c9858ade9b74b45d0/gradle/lib/dependabot/gradle/file_fetcher.rb#L130-L135

Reply:
I'm more of a Maven guy, but indeed, this solution sounds more logical and aligned with the existing implementation, if we only ignore the File Not Found from submodule

Reply:
Fixed!